### DISCUSSION TITLE
Need to disable jsx-a11y rules so that I can use the render prop without warnings, in some cases.

### DISCUSSION THREAD

**Discussion by michaelsawyers** [ariakit/ariakit#3842] [Questions] [answered]
I am using the jsx-a11y plugin, and it's giving me warnings when I do the following:

- `<Role.span render={<h1 />}>Heading</Role.span>` warns `jsx-a11y/heading-has-content`
- `Button render={<a href="/foo" />}>Link</Button>` warns `jsx-a11y/anchor-has-content`

It feels wrong to have to disable these rules, while using Ariakit which is a library that aims to follow a11y rules. 

Is there a better way for me to go about this without disabling a11y lint rules?

**Comment by diegohaz** [comment, marked as answer]
> It feels wrong to have to disable these rules, while using Ariakit which is a library that aims to follow a11y rules.

Ariakit is not violating any a11y rules with that pattern. It seems the jsx-a11y plugin lacks customization options to support it.

I would disable those lint rules for these lines. If you prefer not to, you can pass the children directly to the rendered element:

```jsx
<Role.span render={<h1>Heading</h1>} />
```
